src/main/java/dekk/pw/pokemate/tasks/EvolvePokemon.java

```diff
@@ -34,14 +35,16 @@
     public void run(Context context) {
         try {
             for (Pokemon pokemon : context.getApi().getInventories().getPokebank().getPokemons()) {
-                int number = pokemon.getPokemonId().getNumber();
-                if (CANDY_AMOUNTS.containsKey(number)) {
-                    int required = CANDY_AMOUNTS.get(number);
-                    if (required < 1) continue;
-                    if (pokemon.getCandy() >= required) {
-                        EvolutionResult result = pokemon.evolve();
-                        if (result.isSuccessful()) {
-                            System.out.println(pokemon.getPokemonId() + " has evolved into " + result.getEvolvedPokemon().getPokemonId() + " costing " + required + " candies");
+                if (Config.isWhitelistEnabled() ? Config.getWhitelistedPokemons().contains(pokemon.getId()) : true) {
```

Comment:
What is the result of pokemon.getId()? Is that the unique id or is it the pokemon's number?